llama-index-integrations/embeddings/llama-index-embeddings-deepinfra/llama_index/embeddings/deepinfra/base.py

```diff
@@ -0,0 +1,200 @@
+
+        """
+        url = self.get_url()
+        chunked_data = _chunk(data)
+        embeddings = []
+        for chunk in chunked_data:
+            async with aiohttp.ClientSession() as session:
+                async with session.post(
+                    url,
+                    json={
+                        "inputs": chunk,
+                    },
+                    headers={
+                        "Authorization": f"Bearer {self._api_token}",
+                        "Content-Type": "application/json",
+                    },
+                ) as resp:
+                    response = await resp.json()
+                    embeddings.extend(response["embeddings"])
+        return embeddings
+
+    def _get_query_embedding(self, query: str) -> List[float]:
+        """
+        Get query embedding.
+        """
+        return self._post(self._add_query_prefix([query]))[0]
+
+    async def _aget_query_embedding(self, query: str) -> List[float]:
+        """
+        Async get query embedding.
+        """
+        response = await self._apost(self._add_query_prefix([query]))
+        return response[0]
+
+    def _get_query_embeddings(self, queries: List[str]) -> List[List[float]]:
+        """
+        Get query embeddings.
+        """
+        return self._post(self._add_query_prefix(queries))
+
+    async def _aget_query_embeddings(self, queries: List[str]) -> List[List[float]]:
+        """
+        Async get query embeddings.
+        """
+        return await self._apost(self._add_query_prefix(queries))
+
+    def _get_text_embedding(self, text: str) -> List[float]:
+        """
+        Get text embedding.
+        """
+        return self._post(self._add_text_prefix([text]))[0]
+
+    async def _aget_text_embedding(self, text: str) -> List[float]:
+        """
+        Async get text embedding.
+        """
+        response = await self._apost(self._add_text_prefix([text]))
+        return response[0]
+
+    def _get_text_embeddings(self, texts: List[str]) -> List[List[float]]:
+        """
+        Get text embedding.
+        """
+        return self._post(self._add_text_prefix(texts))
+
+    async def _aget_text_embeddings(self, texts: List[str]) -> List[List[float]]:
+        """
+        Async get text embeddings.
+        """
+        return await self._apost(self._add_text_prefix(texts))
+
+    def _add_query_prefix(self, queries: List[str]) -> List[str]:
+        """
+        Add query prefix to queries.
+        """
+        return [self._query_prefix + query for query in queries]
+
+    def _add_text_prefix(self, texts: List[str]) -> List[str]:
+        """
+        Add text prefix to texts.
+        """
+        return [self._text_prefix + text for text in texts]
+
+
+def _chunk(items: List[str]) -> List[List[str]]:
+    """
+    Chunk items into batches of size MAX_BATCH_SIZE.
+    """
+    return [items[i : i + MAX_BATCH_SIZE] for i in range(0, len(items), MAX_BATCH_SIZE)]
```

Comment:
`MAX_BATCH_SIZE` can be a constructor arg, and this function can be a method

Reply:
Isn't this parameter related with deepinfra's backend?